docs/ref/models/database-functions.txt

```diff
@@ -293,3 +293,518 @@ Usage example::
 .. versionchanged:: 1.9
 
     The ability to register the function as a transform was added.
+
+Date Functions
+==============
+
+.. module:: django.db.models.functions.date
+
+.. versionadded:: 1.10
+
+We'll be using the following model in examples of each function::
+
+    class Experiment(models.Model):
+        start_time = models.DateTimeField()
+        end_time = models.DateTimeField(null=True, blank=True)
+
+``Extract``
+-----------
+
+.. class:: Extract(date_expression, lookup_name=None, **extra)
+
+Extracts a component of a date as a number.
+
+Takes a single ``date_expression`` representing a ``DateField`` or
+``DateTimeField`` and a ``lookup_name``, and returns the part of the date
+referenced by ``lookup_name`` as an ``IntegerField``. Django will generally use
+the databases Extract function, so ``lookup_name``\s that aren't explicitly
+referenced below can still be used provided the database supports that value.
+
+Given the datetime `2015-06-15 14:30:50.321` the ``lookup_name``\s supported by
+Django will return:
+
+* "year": 2015
+* "month": 6
+* "day": 15
+* "week_day": From 1 (Sunday) to 7 (Saturday)
+* "hour": 14
+* "minute": 30
+* "second": 50
+
+.. admonition:: week_day considerations
+
+    The ``week_day`` ``lookup_type`` is calculated differently from most
+    databases and from Python standard functions. This function will return
+    ``1`` for a Sunday, ``2`` for a Monday, through to ``7`` for a Saturday.
+
```

Comment:
The calculation was committed back in the early days (I looked it up, but can't find the commit just now) by Malcolm. There doesn't appear to be a very good reason other than "make all databases return the same value". I was thinking of adding an IsoWeekDay in the future, but it might be a good idea to add it now, and reference that from the docs here. Unfortunately, we'd be stuck with the original being registered as a week_day transform.